homeassistant/components/update/__init__.py

```diff
@@ -322,6 +322,15 @@ def title(self) -> str | None:
         """
         return self._attr_title
 
+    @property
+    def supported_features_compat(self) -> UpdateEntityFeature:
+        """Return the supported features as UpdateEntityFeature.
+
+        Remove this compatibility shim in 2025.1 or later.
+        """
+        features = self.supported_features
+        return UpdateEntityFeature(features) if type(features) is int else features  # noqa: E721
```

Comment:
Same question as in https://github.com/home-assistant/core/pull/106529#discussion_r1437465323

Without a warning, it is probably hard for a custom integration author to identify that this is deprecated.